File: sudoq-app/sudoqmodel/src/main/java/de/sudoq/model/solverGenerator/solution/LockedCandidatesDerivation.java
package de.sudoq.model.solverGenerator.solution;

import java.util.BitSet;
import de.sudoq.model.solvingAssistant.HintTypes;
import de.sudoq.model.sudoku.Constraint;

/**
 * Created by timo on 04.10.16.
 */
public class LockedCandidatesDerivation extends SolveDerivation {


    private Constraint lockedConstraint;
    private Constraint reducibleConstraint;
    private BitSet removableNotes;
    private int note;

    public LockedCandidatesDerivation() {
        super(HintTypes.LockedCandidatesExternal);
    }

    public void setLockedConstraint(Constraint c1){ lockedConstraint = c1; }

    public void setReducibleConstraint(Constraint c1){ reducibleConstraint = c1; }

    public Constraint getReducibleConstraint() {
        return reducibleConstraint;
    }

    public Constraint getLockedConstraint() {
        return lockedConstraint;
    }



    public void setRemovableNotes(BitSet i){
        removableNotes = i;
        note = removableNotes.nextSetBit(0);
    }

    public int getNote() {
        return note + 1;
    }

}

File: sudoq-app/sudoqmodel/src/main/java/de/sudoq/model/solverGenerator/solution/LockedCandidatesDerivation.kt
package de.sudoq.model.solverGenerator.solution

import de.sudoq.model.solvingAssistant.HintTypes
import de.sudoq.model.sudoku.Constraint
import java.util.*

/**
 * Created by timo on 04.10.16.
 */
class LockedCandidatesDerivation : SolveDerivation(HintTypes.LockedCandidatesExternal) {
    var lockedConstraint: Constraint? = null
    var reducibleConstraint: Constraint? = null
    private var removableNotes: BitSet? = null
    private var note = 0
    fun setRemovableNotes(i: BitSet?) {
        removableNotes = i
        note = removableNotes!!.nextSetBit(0)
    }

    fun getNote(): Int {
        return note + 1
    }
}